rust/Earthfile

```diff
@@ -61,11 +66,17 @@ CARGO:
     END
     RUN --mount=$EARTHLY_RUST_CARGO_HOME_CACHE --mount=$EARTHLY_RUST_TARGET_CACHE \
       set -e; \
+      echo "+CARGO: cargo $args"; \
       cargo $args; \
+      if [ "$EARTHLY_DO_NOT_CACHE_PATH_BASED_DEPENDENCIES" = "true" ]; then \
+          cargo metadata --format-version=1  --no-deps | jq -r '.packages[] | select(.id | startswith("path+file://")) | .name' | xargs -I{} cargo clean -p {}; \
```

Comment:
This is too complicated and doesn't match "path" deps like it claims, it's just matching all local crates in an complicated way (since it is matching the crate location not something to do with the dep tree). It could be replaced with `.packages[].name` which is exactly the same due to `--no-deps`.

I think the path based dependencies thing is a red-herring, since we'd want to not cache any local crate since even the crate being built could suffer the false target cache hit problem, not only the things it depends on.

If the principal here is one we want to go fortward with then I'll update to simplify and rename/reword things.